src/set.rs

```diff
@@ -0,0 +1,188 @@
+#![doc = include_str!("../doc/set.md")]
+#![cfg(feature = "alloc")]
+
+#[cfg(not(feature = "std"))]
+use alloc::vec;
+use core::ops;
+
+use wyz::comu::{
+	Const,
+	Mut,
+};
+
+use crate::{
+	boxed::BitBox,
+	order::{
+		BitOrder,
+		Lsb0,
+	},
+	ptr::BitPtr,
+	slice::BitSlice,
+	store::BitStore,
+	vec::BitVec,
+};
+
+mod api;
+mod iter;
+mod traits;
+
+pub use iter::Range;
+
+#[repr(transparent)]
+#[doc = include_str!("../doc/set/BitSet.md")]
+pub struct BitSet<T = usize, O = Lsb0>
+where
+	T: BitStore,
+	O: BitOrder,
+{
+	inner: BitVec<T, O>,
+}
+
+/// Constructors.
+impl<T, O> BitSet<T, O>
+where
+	T: BitStore,
+	O: BitOrder,
+{
+	/// An empty bit-set with no backing allocation.
+	pub const EMPTY: Self = Self {
+		inner: BitVec::EMPTY,
+	};
+
+	/// Creates a new bit-set for a range of indices.
+	#[inline]
+	pub fn from_range(range: ops::Range<usize>) -> Self {
```

Comment:
This felt like the best analogue for `BitVec::repeat` to me. Since only allowing `RangeFrom` seemed weird, I decided to accept a proper `Range` instead.